libindy/src/services/pool/pool.rs

```diff
@@ -308,7 +309,8 @@ impl<T: Networker, R: RequestHandler<T>> PoolSM<T, R> {
                     PoolEvent::CatchupRestart(merkle_tree) => {
                         if let Ok((nodes, remotes)) = _get_nodes_and_remotes(&merkle_tree) {
                             state.networker.borrow_mut().process_event(Some(NetworkerEvent::NodesStateUpdated(remotes)));
-                            state.request_handler = R::new(state.networker.clone(), _get_f(nodes.len()), &vec![], &nodes, None, &pool_name, timeout, extended_timeout);
+                            let generator : Generator = Generator::from_bytes(&DEFAULT_GENERATOR.from_base58().unwrap()).unwrap();
```

Comment:
I don't like the idea to duplicate this one-line logic all the time.
I suggest to pick one of the options:
1) return back optional parameter but make name of the parameter be more concrete that in majority of cases (all cases for now) default one will be used. Like "specififc_generator"  - optional generator otherwise default one will be used
2) eliminate this parameter completely and just setup default generator in `RequestSM::new`

Reply:
Do you know why the constant has the value it has? Where does that come from?

I have changed the code to use random generators and the tests still pass. So what is the point of this constant. Also it seems the generators not even have to be the same. Strange.

Is something wrong with the signature validation?

Reply:
It should be the same as the generator on the Ledger side. Most probably tests are passed due to fallback path of read requests. To double-check it manually you can run any ledger read request and look for state proof validation status in trace log

Reply:
Just pushed code implementing option 2